### ISSUE TITLE
Introduce types of taxonomies

### ISSUE THREAD

**Issue by gallypette** [MISP/MISP#2159]
I sum up here the argument we had for adding types of taxonomies. Basically it would give a separation of concern for taxonomies. Some taxonomies could be applied only to event, others to users, etc.
The main purpose of this is to support new features.

- event: what taxonomies are about ATM.
- org: taxonomies about organisations (fields of expertise for instance.)
- user: taxonomies about users (experience, fields of expertise for instance.)
- badges: taxonomies used for the development of gamification features (achievements.)

**Comment by gallypette** [comment]
I think it could be useful to also be able to extend the type constraint to attribute's categories.
For instance a type "attr:external-analysis" or "attr:analysis" would be very useful for building a taxonomies regarding how an analysis has been conducted but would have nothing to do with "attr:network-activity".

**Comment by iglocska** [comment]
Makes sense, though for more efficient parsing I'd use a separate key for
it.
